Dockerfile

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 the Velero contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+FROM --platform=$BUILDPLATFORM golang:1.14 as builder-env
+
+ARG PKG
+ARG VERSION
+ARG GIT_SHA
+ARG GIT_TREE_STATE
+
+ENV CGO_ENABLED=0 \
+    GO111MODULE=on \
+    GOPROXY=https://proxy.golang.org \
```

Comment:
Is there any way to overide this variable in at build time? I ask because there are times that Velero will be built within a private environment, without access to the internet, and an internal goproxy server should be used.

Perhaps we introduce another `ARG`?

https://vsupalov.com/docker-build-pass-environment-variables/ also lets us use the hosts's GOPROXY env var value at build time, which I think would be useful as a shortcut, but I think making it explicit is easier to follow later on.

Reply:
@nrb fair point. I just pushed a commit which makes `GOPROXY` for both the regular build-image as the multi-arch build image configurable with the default set to `https://proxy.golang.org`.